### PR TITLE
Feat e2e android file upload

### PR DESCRIPTION
**Pull request by Karim94**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/1105/mobile-android-tests-e2e-pick-an-image-from-gallery
#### Testing instructions  

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

**Comment by seavan** [comment]
@alexandra-marin any news here?

**Comment by alexandra-marin** [comment]
I've asked @Karim94 to remove changes not relevant to the task. Waiting on that.

**Comment by seavan** [comment]
@Karim94 bump ^

**Comment by Karim94** [comment]
@seavan @alexandra-marin Done.Sorry for the delay, not sure why I missed Alexandra's comment.

File: app/components/contacts/contact-selector.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ export default class ContactSelector extends SafeComponent {
             alignItems: 'center',
```
Comment (alexandra-marin) [inline app/components/contacts/contact-selector.js]: why is this file changed?

File: app/components/files/files.js [hunk 2]
```diff
@@ -164,18 +168,17 @@ export default class Files extends SafeComponent {
             flexGrow: 1,
             flexDirection: 'row',
             alignItems: 'center',
-            backgroundColor: 'white',
```
Comment (alexandra-marin) [inline app/components/files/files.js]: why is this changed?

File: app/components/layout/tabs.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ export default class Tabs extends SafeComponent {
         ) : null;
         return (
             <TouchableOpacity
-                {...testLabel(icon)}
+                {...testLabel(route)}
```
Comment (alexandra-marin) [inline app/components/layout/tabs.js]: route will change depending on state, use icon instead

File: app/components/shared/avatar.js [hunk 4]
```diff
@@ -22,8 +22,6 @@ import { popupSetupVideo } from '../shared/popups';
 import CircleButtonWithIcon from '../controls/circle-button-with-icon';
```
Comment (alexandra-marin) [inline app/components/shared/avatar.js]: why is this file changed?

File: test/code/AndroidFactory.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const { android } = require('./platforms');
```
Comment (alexandra-marin) [inline test/code/AndroidFactory.js]: duplicated file, you can use the one in helpers folder(or remove that one)

File: test/code/buildPaths.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
     iOSBuildPath: `${__dirname}/../../ios/build/Build/Products/Debug-iphonesimulator/peeriomobile.app`,
+    // androidBuildPath: `${__dirname}/../../android/app/build/outputs/apk/app-armeabi-v7a-debug.apk`
```
Comment (alexandra-marin) [inline test/code/buildPaths.js]: why is this changed?

File: test/code/helpers/accountHelper.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+async function loginExistingAccount(app, username, password) {
```
Comment (alexandra-marin) [inline test/code/helpers/accountHelper.js]: use the helpers in world.js instead of these

File: test/code/pages/actionSheetsPage/AndroidChatActionSheetPage.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+const Page = require('../page');
```
Comment (alexandra-marin) [inline test/code/pages/actionSheetsPage/AndroidChatActionSheetPage.js]: duplicated file. use the one in popups folder

File: test/code/pages/files/AndroidFileUploadPage.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+        // Assumption: 1 album per row, 1 per column
+        const albumsPerRow = 1;
+        const albumsPerColumn = 1;
+
+        // Tap in the center of first album
+        await this.app.touchPerform([{
+            action: 'tap',
+            options: {
+                x: galleryX + (galleryW / albumsPerRow) / 2,
```
Comment (alexandra-marin) [inline test/code/pages/files/AndroidFileUploadPage.js]: don't need such complex calculations if you just tap in the middle of container.
move everything in page objects (if possible)

File: app/components/contacts/contact-selector.js (hunk 1 above)
Comment (Karim94) [reply app/components/contacts/contact-selector.js]: I don't see any changes made to `contact-selector.js`, I will remove the unnecessary space deleted in `create-channel.js`.

File: test/code/buildPaths.js (hunk 6 above)
Comment (Karim94) [reply test/code/buildPaths.js]: The commented buildPath would not work for me, and since everyone is using similar technologies for android (Genymotion), I would assume the same for others. I kept the old one for reference.

**Comment by seavan** [comment]
@alexandra-marin @Karim94 conflicts must be resolved

**Comment by Karim94** [comment]
@alexandra-marin @seavan I'll do it soon.

**Comment by Karim94** [comment]
@seavan merge is done

**Comment by seavan** [comment]
fails on iOS:

image

@alexandra-marin does it work for you on iOS?

**Comment by Karim94** [comment]
@seavan These tests aren't supposed to work on iOS, hence the name feat-e2e-**android**-file-upload :smile:

**Comment by alexandra-marin** [comment]
I made the changes for iOS, that should work. They’re failing because of the scroll problem.

**Comment by seavan** [comment]
waiting for @alexandra-marin fixes for iOS 11.2 gallery and for @Karim94 fixes for multiple images gallery.
